starfish/image/_filter/util.py

```diff
@@ -111,15 +111,20 @@ def validate_and_broadcast_kernel_size(
     return valid_sigma
 
 
-def preserve_float_range(array: Union[xr.DataArray, np.ndarray]) -> Union[xr.DataArray, np.ndarray]:
+def preserve_float_range(
+        array: Union[xr.DataArray, np.ndarray],
+        rescale: bool=False) -> Union[xr.DataArray, np.ndarray]:
     """
-    Clip values below zero and scale values by the max, but only if sub- or super-maximal values are
-    detected. Otherwise, this function will preserve the dynamic range of the input image.
+    Clip values below zero and and above one or scale values by the max,
```

Comment:
I think this reads a bit confusingly, since the `or` clause doesn't have a clean antecedent. 

I would rephrase this as 
```
Clips values below zero. If values above one are detected, clips them to 1 unless `rescale` is True, 
in which case the input is scaled by the max value and the dynamic range is preserved. 
```